File: ruby_project/Task_scripts/Text_Dollar.rb
begin	
class NumberToWord
	def convert(number)
		hash_units = { 1000000 => "Million", 1000 => "Thousand", 100 => "Hundred",  90 => "Ninety", 80 => "Eighty",  70 => "Seventy",  60 => "Sixty", 50 => "Fifty",  40 => "Forty",  30 => "Thirty", 20 => "Twenty", 19=>"Nineteen", 18=>"Eighteen", 17=>"Seventeen", 16=>"Sixteen", 15=>"Fifteen",   14=>"Fourteen",  13=>"Thirteen",  12=>"Twelve",  11 => "Eleven", 10 => "Ten", 9 => "Nine", 8 => "Eight", 7 => "Seven", 6 => "Six", 5 => "Five",  4 => "Four", 3 => "Three",  2 => "Two", 1 => "One"}
		raise  if (number =~ /[a-z]/im )
		number = number.to_i
		raise "Input is not right" if number<0 
		hash_units.each { |key, value|
			if number == 0
				return " "
			elsif number < 10 
				return "#{hash_units[number]}"      
			elsif number < 100 && number / key > 0
				return "#{value}" if number % key == 0
				return "#{value}" + convert(number % key)
			elsif number / key > 0
				return convert(number / key) + "#{value}" + convert(number % key)
			end
		}
		
	end
end
	n2w = NumberToWord.new
	ARGV.each { |number|
		begin
			puts "#{n2w.convert(number)}Dollars" 
		rescue 
			puts "invalid input"
			next
		end
	}
rescue
	puts "Invalid input"
end

File: Java_project/Task_codes/TextDollar.java
import java.io.*;
import java.util.*;
class TextDollar
{	static int[] numbers = {1000000, 1000, 100, 90, 80, 70, 60, 50, 40, 30, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1};
	static String[] text = {"Million","Thousand","Hundred","Ninety","Eighty","Seventy","Sixty","Fifty","Forty","Thirty","Twenty","Nineteen","Eighteen","Seventeen","Sixteen","Fifteen","Fourteen","Thirteen","Twelve",
"Eleven","Ten","Nine","Eight","Seven","Six","Five","Four","Three","Two","One"};
	public static void main (String[] args) throws IOException
	{
		String line;
		for(int index = 0; index<args.length;index++) {
			line = args[index].trim();
			if (line.length() == 0)
			{
				System.out.println("Blank Line!!");
				continue;
			}
			try{
			int number = Integer.parseInt(line);
			System.out.println(convert(number)+"Dollars");
			} catch(Exception e){System.out.println("invalid input");continue;}
			
		}
	}
	public static String convert(int number){
		for(int i=0;i<numbers.length;i++){
			if(number == 0){
				return " ";
			}
			else if(number < 10){
				return text[text.length - number];
			}
			else if(number < 100 && number/numbers[i] > 0){
				if(number % numbers[i] == 0)
					return text[i];
				return text[i]+convert(number % numbers[i]);
			}
			else if(number/numbers[i] > 0){
				return convert(number / numbers[i]) + text[i] + convert(number % numbers[i]);
			}
	
		}
		return "";
	}
}
